### DISCUSSION TITLE
Ошибка с Google DIsk+История прослушивания+Триггер

### DISCUSSION THREAD

**Discussion by efremych** [Chimildic/goofy#210] [Помощь с ошибкой]
Я понимаю, что возможно мало информации.
Но, может, поможите с направлением хотя бы.

Вопросы по скрину:
1. Почему при запуске вручную ошибки нет (последняя функция)
2. Почему номера строк в логе не совпадают с библиотекой, как найти бы хотя бы нужные места (хотя конечно примерно понимаю где они)...
![image](https://user-images.githubusercontent.com/76212904/189416229-d6992240-bba2-4bd7-b687-95dbe4a2aa3f.png)

Ошибка пораждает следующее:
Создаются новые файлы с десятком-дюжиной треков
![image](https://user-images.githubusercontent.com/76212904/189416845-399f3a53-8f54-40ba-a089-440fc185d498.png)

Вот так это выглядит
![image](https://user-images.githubusercontent.com/76212904/189416996-5c4f1097-2dcf-4688-b42c-5c36c21ffbbf.png)

- GetId я использую для проверки размера файла
- последнее действие что делал с Goofy - обновлял права доступа на днях (была почему-то ошибка 401)
- ксттаи, да, оказывается ровно после обновления прав началось
- 
![image](https://user-images.githubusercontent.com/76212904/189417337-5a8d9b64-df47-43c4-9558-f6be06edbf50.png)

Как это всё может быть связано?...
Может я как-то неправильно права обновил?

**Comment by efremych** [comment]
так и должно быть? :)
Версия 4
![image](https://user-images.githubusercontent.com/76212904/189418403-6a1f1fa4-3c21-4887-8490-76bab8e3b43d.png)

**Comment by Chimildic** [comment]
Исходя из ошибки, `getId` вызывается на пустом объекте. То есть истинная ошибка происходит раньше, при получении объекта файла. По какой-то причине искомый файл не находится. 

`updateRecentTracks_` нужно удалить. Автоудаление триггера вообще заложено, но либо оно не отработало, либо у вас код не целиком обновлён.

**Reply by efremych** [reply]
> Исходя из ошибки, getId вызывается на пустом объекте

Так в том то и дело, что вроде нет. Ручным запуском функции файл подтягивался.

>updateRecentTracks_ нужно удалить. Автоудаление триггера вообще заложено, но либо оно не отработало, либо у вас код не целиком обновлён.

До обновления прав доступа всё было тихо, один триггер :)
![image](https://user-images.githubusercontent.com/76212904/189421643-5b38f149-45ce-4440-915e-ca347ac89d30.png)

Мистика какая-то. 
И главное, почему в логах ошибки, какие-то левые номера строк библиотеки. Как будто из какой-то другой версии библиотеки (старой что ли или типа того)

Так, ну вроде удалился триггер, получается
![image](https://user-images.githubusercontent.com/76212904/189422011-9fd3ab40-6c74-4484-8192-583f5560d288.png)

И нотификаций об ошибках никаких на почту не приходило. Странно...

В общем, склоняюсь к ошибке обновлений прав доступа. Может, не то нажал, так как крайне редко это делается, а в доках пример из новой версии редактора (который я пока пытаюсь избегать :)), надо было через него делать.

**Reply by Chimildic** [reply]
"Версия 4" как раз слепок кода, который отличается от "версии разработчика" (нынешний код без сохранения развёртывания). Отсюда разные номера строк. 

Удалите триггер от версии 4 и понаблюдайте. У остальных триггеров также не должно быть жёстко зашитых версий.

**Reply by efremych** [reply]
Жесть. Спасибо. Удалил, да.
